overlord/hookstate/hooks.go

```diff
@@ -176,9 +182,14 @@ func (h *gateAutoRefreshHookHandler) Done() (err error) {
 			return err
 		}
 		if h.refreshAppAwareness {
-			// we have HintInhibitedGateRefresh lock already when running the hook,
-			// change it to HintInhibitedForRefresh.
-			if err := runinhibit.LockWithHint(snapName, runinhibit.HintInhibitedForRefresh); err != nil {
+			// we have HintInhibitedGateRefresh lock already when running the hook, change
+			// it to HintInhibitedForRefresh.
+			// Also let's reuse inhibit info that was saved in Before().
+			_, inhibitInfo, err := runinhibit.IsLocked(snapName)
```

Comment:
here

Reply:
Nice catch, This was actually done to reuse what is already set in the `Before()` hook as discussed [here](https://github.com/snapcore/snapd/pull/13174#discussion_r1318493024).